### PR TITLE
Handle and report manifest errors and provide a fallback

### PR DESCRIPTION
**Pull request by jlks**
📺 What

Currently there is no way to know manifest parsing has failed. This PR provides a plugin hook `onManifestParseError` and explicitly defines the fallback behaviour of ManifestParser.

> Tickets: N/A

🛠 How

- Add `onManifestParseError` to the plugin interface.
- Change returning undefined from `convertEpochMsToVideoTimeSeconds` and `convertVideoTimeSecondsToEpochMs` to return null.

### CODE REVIEW

File: src/pluginenums.js [hunk 1]
```diff
@@ -10,6 +10,7 @@ export default {
     ERROR: 'error'
   },
   ERROR_CODES: {
+    MANIFEST_PARSE: 7,
     BUFFERING_TIMEOUT: 8,
     MANIFEST: 9
```
Comment (matt-stephenson) [inline src/pluginenums.js:15]: Could we change this to MANIFEST_LOAD?

File: src/manifest/manifestparser.js [hunk 2]
```diff
@@ -26,26 +29,34 @@ function parseMPD (manifest, dateWithOffset) {
         correction: timeCorrection
       }
     } else {
-      return { error: 'Error parsing DASH manifest attributes' }
+      throw new Error('manifest-dash-attributes-parse-error')
     }
   } catch (e) {
-    return { error: 'Error parsing DASH manifest' }
+    const error = new Error(e.message || 'manifest-dash-parse-error')
```
Comment (Luke-Chatburn) [inline src/manifest/manifestparser.js:35]: Will we not get a different message here if it actually fails on the try/catch? Do we not want it to always be either 'manifest-dash-parse-error' or 'manifest-dash-attributes-parse-error'

Also, we only ever report the single code and we have multiple paths of errors here? do we want to report this extra information?

File: src/pluginenums.js (hunk 1 above)
Comment (matt-stephenson) [reply src/pluginenums.js:15]: Sorry, maybe GitHub formatting, I mean the one on line 15, that's currently just MANIFEST.

**Review by Luke-Chatburn — APPROVED** [review]

File: src/manifest/manifestparser.js (hunk 2 above)
Comment (jlks) [reply src/manifest/manifestparser.js:35]: Yeah you end up with the different message but the same code, both of which are reported to the plugin. It is a bit annoying having to rely on the error throwing upwards.

**Pull request merged**